tests/coverage-map/if.cov-map

```diff
@@ -0,0 +1,13 @@
+Function name: if::main
+Raw bytes (28): 0x[01, 01, 02, 01, 05, 05, 02, 04, 01, 04, 01, 02, 0c, 05, 02, 0d, 02, 06, 02, 02, 06, 00, 07, 07, 01, 05, 01, 02]
+Number of files: 1
+- file 0 => global file 1
+Number of expressions: 2
+- expression 0 operands: Counter(0), Counter(1)
+- expression 1 operands: Counter(1), Expression(0, Sub)
+Number of file 0 mappings: 4
+- Code(Counter(0)) @ (prev + 4, 1) to (start + 2, 12)
+- Code(Counter(1)) @ (prev + 2, 13) to (start + 2, 6)
+- Code(Expression(0, Sub)) @ (prev + 2, 6) to (start + 0, 7)
+- Code(Expression(1, Add)) @ (prev + 1, 5) to (start + 1, 2)
```

Comment:
Some explanation of how to read these mappings might be good.
Intuitively, I would assume that `Expression` should be read as "$expression_index, $operator joining expression operands".
So this would give us:
- `Counter(0) - Counter(1)`, intuitively makes sense, as this is basically an empty region for the non-existant `else` block
- `Counter(1) + (Counter(0) - Counter(1))` = `Counter(0)`, as this is just the code block outside the if/else.

I’m wondering if this could be simplified in a followup, as the second expression could just be `Counter(0)`.

Reply:
My primary goal here is to present the mapping data exactly as it appears in the file. When we currently emit expressions that are unnecessarily complex, I want that to be visible in the snapshots.

That said, I do think it could be useful to print a more readable summary of each expression mapping, on a separate line. Trying to interpret them manually gets old pretty quickly.

Reply:
And yeah, at some point I do want to change the instrumentor/codegen code to reduce the number of redundant expression nodes it emits. The existing code doesn't give this much attention, and for historical reasons it has a habit of producing `(x + 0)` expressions that shouldn't need to exist.

However, I want to leave that until after I'm done with my current refactoring plans, because those should make the necessary changes a fair bit easier.

Reply:
I updated the dump tool to also print a "resolved" form of each expression that appears directly in a mapping.